common/src/main/java/com/sequenceiq/cloudbreak/vm/VirtualMachineConfiguration.java

```diff
@@ -1,21 +1,24 @@
 package com.sequenceiq.cloudbreak.vm;
 
-import java.util.List;
+import java.util.Arrays;
+import java.util.Set;
+import java.util.stream.Collectors;
 
-import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Component;
 
 @Component
-@ConfigurationProperties("vm")
 public class VirtualMachineConfiguration {
 
-    private List<Integer> supportedJavaVersions;
+    private final Set<Integer> supportedJavaVersions;
 
-    public List<Integer> getSupportedJavaVersions() {
-        return supportedJavaVersions;
+    public VirtualMachineConfiguration(@Value("${vm.supportedJavaVersions}") String supportedJavaVersions) {
```

Comment:
Could you please cover this with UT? A regular UT is sufficient, no need for a real Spring `ApplicationContext`. :smile:

Reply:
Sure, done